src/main/java/ortus/boxlang/runtime/net/BoxHttpClient.java

```diff
@@ -0,0 +1,2461 @@
+
+			// Populate standard response metadata
+			HttpResponseHelper.populateResponseMetadata( this.httpResult, headers, httpVersionString, response.statusCode() );
+
+			// Process Content-Type header and extract charset for streaming
+			String	contentType		= HttpResponseHelper.extractFirstHeaderByName( headers, Key.contentType );
+			String	charset			= HttpResponseHelper.processContentType( this.httpResult, headers, contentType, this.charset );
+			String	contentEncoding	= HttpResponseHelper.extractFirstHeaderByName( headers, Key.contentEncoding );
+
+			// Determine if SSE mode (auto-detect or forced)
+			boolean	isSSE			= this.forceSSE || ( contentType != null && contentType.contains( "text/event-stream" ) );
+
+			// Execution Markers
+			this.httpResult.put( Key.stream, true );
+			this.httpResult.put( Key.sse, isSSE );
+
+			/**
+			 * ------------------------------------------------------------------------------
+			 * ON HTTP RAW RESPONSE EVENT
+			 * ------------------------------------------------------------------------------
+			 */
+			interceptorService.announce(
+			    BoxEvent.ON_HTTP_RAW_RESPONSE,
+			    ( java.util.function.Supplier<IStruct> ) () -> Struct.ofNonConcurrent(
+			        Key.result, this.httpResult,
+			        Key.response, response,
+			        Key.httpClient, BoxHttpClient.this,
+			        Key.httpRequest, this.targetHttpRequest
+			    ) );
+
+			/**
+			 * ------------------------------------------------------------------------------
+			 * PROCESS STREAMING RESPONSE BY MODE
+			 * ------------------------------------------------------------------------------
+			 */
+			if ( isSSE ) {
+				processSSEStream( response, headers, charset, contentEncoding );
+			} else {
+				processRegularStream( response, headers, charset, contentEncoding );
+			}
+		}
+
+		/**
+		 * Process an SSE (Server-Sent Events) stream
+		 *
+		 * @param response        The HTTP response with InputStream body
+		 * @param headers         The response headers struct
+		 * @param charset         The charset to use for reading
+		 * @param contentEncoding The content encoding (ignored for SSE, always text)
+		 *
+		 * @throws IOException If an I/O error occurs
+		 */
+		private void processSSEStream(
+		    HttpResponse<java.io.InputStream> response,
+		    IStruct headers,
+		    String charset,
+		    String contentEncoding ) throws IOException {
+
+			// Logging essential for SSE processing
+			BoxLangLogger streamLogger = getLogger();
+			streamLogger.trace( "Starting SSE stream processing. Charset: {}", charset );
+
+			// Prepare for SSE processing
+			SSEParser		parser			= new SSEParser();
+			StringBuilder	accumulatedData	= new StringBuilder();
+			AtomicLong		eventCount		= new AtomicLong( 0 );
+			AtomicBoolean	streamingError	= new AtomicBoolean( false );
+
+			// Read SSE stream (always UTF-8 text, no decoding)
+			// Try with Resources to ensure streams are closed properly
+			try (
```

Comment:
I did not know you could do try-with-resources with multiple resources like this. Nice!